### PR TITLE
issue/1229 – Make license key configurable via wp-config.php

### PR DESCRIPTION
**Pull request by DrewAPicture**
Issue: #1229 

Introduce a new constant, `AFFILIATEWP_LICENSE_KEY`, which can be defined in wp-config.php to override the license for all sites in a network or simply override the license key in a single site.

The new constant is checked in one place, a new static method in `Affiliate_WP_Settings`, `get_license_key()`.

### CODE REVIEW

**Comment by pippinsplugins** [comment]
@DrewAPicture looks like we need to bring this / 1.9 up to date with master as well.

File: includes/admin/settings/class-settings.php [hunk 1]
```diff
@@ -358,21 +371,14 @@ function get_registered_settings() {
...
-			$debug_mode_desc = __( 'Check this box to enable debug mode. This will turn on error logging for the referral process to help identify problems.', 'affiliate-wp' );
-		}
+		/**
+		 * Fires before attempting to retrieve registered settings.
+		 *
+		 * @since 1.9
+		 *
+		 * @param Affiliate_WP_Settings $this Settings instance.
+		 */
+		do_action( 'pre_get_registered_settings', $this );
```
Comment (pippinsplugins) [inline includes/admin/settings/class-settings.php]: Shouldn't this be prefixed with `affwp_` or is there a reason for using the same action as WP core?

Comment (DrewAPicture) [inline includes/admin/settings/class-settings.php]: @pippinsplugins Nope, that's a good call. I also thought about it a little bit, and if we don't want to open up an entry point for third-party developers with this hook, we could also just manually fire those two callbacks at the point of the hook. Thoughts?

Comment (pippinsplugins) [inline includes/admin/settings/class-settings.php]: I like having the hooks.

Comment (DrewAPicture) [inline includes/admin/settings/class-settings.php]: okie dokie!

**Pull request merged**